src/app/features/rpc-transaction-request/use-rpc-transaction-request.tsx

```diff
@@ -9,23 +9,47 @@ import { analytics } from '@shared/utils/analytics';
 import { focusTabAndWindow } from '@app/common/focus-tab';
 import { useRpcRequestParams } from '@app/common/hooks/use-rpc-request-params';
 
-export interface RpcTransactionRequest {
-  origin: string;
-  requestId: string;
-  tabId: number;
+interface TransactionState {
   isBroadcasting: boolean;
   isLoading: boolean;
   isSubmitted: boolean;
+}
+
+type TransactionAction =
+  | { type: 'SET_BROADCASTING'; payload: boolean }
+  | { type: 'SET_LOADING'; payload: boolean }
+  | { type: 'SET_SUBMITTED'; payload: boolean };
+
+const initialState: TransactionState = {
+  isBroadcasting: false,
+  isLoading: false,
+  isSubmitted: false,
+};
+
+function transactionReducer(state: TransactionState, action: TransactionAction): TransactionState {
+  switch (action.type) {
+    case 'SET_BROADCASTING':
+      return { ...state, isBroadcasting: action.payload };
+    case 'SET_LOADING':
+      return { ...state, isLoading: action.payload };
+    case 'SET_SUBMITTED':
```

Comment:
Can you explain these states as well? What's the difference between broadcasting and loading? Submitted is the form has been submitted?

Reply:
Loading state loads data (like balances) and broadcasting is for when the button is clicked to broadcast the transaction. This was setup in `getTransactionActions` by @alter-eggo when `sendTransfer` was built. I am trying to work with that for now bc in the next UI updates issue we will have to change quite a lot implementing the flows for the different states of the transaction. Design team has some flows for pending, confirmed, and failed transactions: https://www.figma.com/design/NbCzd27zkHOuhOgWD0x0nw/%F0%9F%91%8D-Product-area--Approver-view?node-id=29404-64642&t=Okitmwnm0MxghRPI-0